src/test-support/mocks/src/hostname-generators/_/_overview.ts

```diff
@@ -0,0 +1,49 @@
+import type { HostnameGeneratorItem } from '@/app/hostname-generators/data'
+import type { EndpointDependencies, MockResponder } from '@/test-support'
+
+export default ({ fake, env }: EndpointDependencies): MockResponder => (req) => {
+  const { params } = req
+  const name = params.name as string
+  const [displayName = '', namespace = ''] = name.split('.')
```

Comment:
Ah! I really like this form. I think we could use this everywhere in the mocks to replace the horrible `.pop`ing thing I've been doing 😆  🎉  Nice!